File: Models/DTO/EventDTO.cs

namespace SportBin.Models.DTO
{
    public class EventDTO : BaseDTO
    {
        public string TeamOneName { get; set; }
        public string TeamTwoName { get; set; }
        public int TeamOneScore { get; set; }
        public int TeamTwoScore { get; set; }
        public string ShortDescription { get; set; }
        public string Description { get; set; }
        public DateTime Date { get; set; }
        public ICollection<string> PhotoUrls { get; set; }
        public ICollection<string> CategoryNames { get; set; }
    }
}

File: ClientApp/src/app/models/DTOs/EventDTO.ts
export class EventDTO implements IEventDTO {
  id: string;
  teamOneName: string;
  teamTwoName: string;
  teamOneScore: number;
  teamTwoScore: number;
  shortDescription: string;
  description: string;
  date: Date;
  photoUrls: string[];
  categoryNames: string[];

  constructor(
    id: string,
    teamOneName: string,
    teamTwoName: string,
    teamOneScore: number,
    teamTwoScore: number,
    shortDescription: string,
    description: string,
    date: Date,
    photoUrls: string[],
    categoryNames: string[]
  ) {
    this.id = id;
    this.teamOneName = teamOneName;
    this.teamTwoName = teamTwoName;
    this.teamOneScore = teamOneScore;
    this.teamTwoScore = teamTwoScore;
    this.shortDescription = shortDescription;
    this.description = description;
    this.date = date;
    this.photoUrls = photoUrls;
    this.categoryNames = categoryNames;
  }
}

export interface IEventDTO {
  id?: string;
  teamOneName?: string;
  teamTwoName?: string;
  teamOneScore?: number;
  teamTwoScore?: number;
  shortDescription?: string;
  description?: string;
  date?: Date;
  photoUrls?: string[];
  categoryIds?: string[];
  categoryNames?: string[];
}

export interface IEventCategoryDTO {
  eventId: string;
  categoryId: string;
  event: IEventDTO;
  category: ICategoryDTO;
}

export interface ICategoryDTO {
  id: string;
  name: string;
}
